lib/improver/tests/nbhood/square_kernel/test_SquareNeighbourhood.py

```diff
@@ -890,11 +892,11 @@ def test_masked_array_re_mask_true(self):
                            [0, 0, 1, 1, 0],
                            [0, 0, 1, 1, 0]]]])
         expected_array = np.array(
-            [[[[0.000000, 0.000000, 0.571429, 0.500000, 0.000000],
-               [0.000000, 0.750000, 0.571429, 0.428571, 0.000000],
-               [0.000000, 0.000000, 0.714286, 0.571429, 0.200000],
-               [np.nan, 0.000000, 0.666667, 0.571429, 0.000000],
-               [np.nan, 0.000000, 0.666667, 0.666667, 0.000000]]]])
+            [[[[1.0000, 0.666667, 0.600000, 0.500000, 0.50],
```

Comment:
I think for this test the result is a masked array. The reason you had to change the NaNs for travis is probably because it had a different way of compared masked arrays.
I think we should maybe be asserting `result.data.data` is equal to `expected_array`. And checking the mask separately. The same may apply to other unit tests.

Reply:
Done